content/technotes/2020-07-09-cran-checks-update/index.Rmd

```diff
@@ -0,0 +1,138 @@
+
+<!-- <img src="/img/blog-images/2019-10-09-cran-checks-api-update/svgs/ok.svg"> -->
+
+<br>
+
+### Documentation
+
+APIs are not very useful without good documentation.  Maëlle has single-handedly produced very nice documentation for the API: https://docs.cranchecks.info/
+
+The documentation includes explanation of all the API routes, and includes examples in Shell/command line and for use in R. 
+
+There's also detailed explanation of notifications, see below.
+
+> Maëlle, what do you want to discuss here?
+
+
+<br>
+
+### Notifications
+
+Good technical solutions are often born from scratching one's own itch. The first author has many packages on CRAN and would like to avoid getting emails from the CRAN maintainers with a deadline to fix a problem. If I could only know about a problem with a CRAN check and fix it quickly, we're all better off as users get fixes quickly, and CRAN maintainers email burden is that much less. 
+
+We're announcing here the availability of cran checks notifications. These notificaitons are emails; there could be other forms (e.g., twitter, etc.), but emails probably meet most people's needs. To get started see the docs at <https://docs.cranchecks.info/#notifications>.
+
+Notifications work via a rule that you set. A rule is made up of one or more of four categories:
+
+- status: match against check status. one of: ok, note, warn, error, or fail
+- time: days in a row the match occurs. an integer. can only go 30 days back (history cleaned up after 30 days)
+- platforms: platform the status occurs on, including negation (e.g., “-solaris”). options: solaris, osx, linux, windows, devel, release, patched, oldrel
+- regex: a regex to match against the text of an error in `check_details.output`
+
+A user can set as many rules as they like. A package can have more than one rule. Users can delete only their own rules (as one would expect). Rules are used indefinitely, until deleted by the user. 
+
+Once a rule is triggered an email is sent to the user. We won't send the same email for 5 days. After 5 days has past, if the rule is matched, we'll send the same email; and repeat. 
+
+Feel free to ignore these emails, or act them as you see fit. 
+
+The email is structured as follows:
+
+{{<figure src="email.png" alt="example cran checks email" width="450">}}
+
+- Triggered rule: with a list of each of the four categories and their values.
+- Date of the check result (matches closely the date on the CRAN check results that CRAN maintainers created)
+- Your check results: link to the CRAN checks API JSON output for your package
+- The rest is information, report bugs, ask for help, etc. Note the "Unsubscribe" link doesn't actually work yet!
+
+#### Managing notifications
+
+There is no web interface to notifications at this time. The only interface is the cchecks R package.
```

Comment:
or any other way to interact with the web API e.g. curl?

Reply:
will do